modules/cas_disk/exp_obj.c

```diff
@@ -575,15 +568,14 @@ int casdsk_exp_obj_create(struct casdsk_disk *dsk, const char *dev_name,
 
 error_set_geometry:
 	blk_cleanup_queue(exp_obj->queue);
-error_init_queue:
-	blk_mq_free_tag_set(&dsk->tag_set);
-error_init_tag_set:
-	_casdsk_exp_obj_clear_dev_t(dsk);
 error_dev_t:
 	put_disk(gd);
 error_alloc_disk:
 	module_put(owner);
 	dsk->exp_obj->owner = NULL;
+	blk_mq_free_tag_set(&dsk->tag_set);
+error_init_tag_set:
+	_casdsk_exp_obj_clear_dev_t(dsk);
```

Comment:
`_casdsk_exp_obj_set_dev_t()` is the last but one function that might return an error therefore `_casdsk_exp_obj_clear_dev_t()` should be called as a rollback for `error_set_geometry` (which is the last error)